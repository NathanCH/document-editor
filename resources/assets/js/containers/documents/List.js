import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './ListItem';
import withLoader from './withLoader';

import './List.scss';

const List = props => (
  <div className="list">
    {props.items.map(item => (
      <ListItem 
        item={item} 
        key={item.id} />
    ))}
  </div>
);

List.propTypes = {
  items: PropTypes.array,
  isFetching: PropTypes.bool,
};

List.defaultProps = {
  items: [],
  isFetching: false,
};

export default withLoader(List);