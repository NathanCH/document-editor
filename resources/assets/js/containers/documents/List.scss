@import './resources/assets/sass/variables';

.list {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-gap: 2rem;
}