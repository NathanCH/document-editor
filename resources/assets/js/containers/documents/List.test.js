import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';

const mockItems = {
  1: {
    id: 1,
    title: 'Mocked document',
    created_at: '',
    updated_at: '',
  },
  2: {
    id: 2,
    title: 'Second mocked document',
    created_at: '',
    updated_at: '',
  },
};

describe('<List />', () => {
  test('it renders without props', () => {
    const component = renderer
      .create(<List />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('it renders a <ListItem /> for each `props.items`', () => {
    const component = renderer
      .create(<List items={mockItems} />);

    let instance = component.root;

    const foundListItems = instance.findAll(
      el => el.type.name == 'ListItem'
    );

   expect(foundListItems).toHaveLength(2);
  });

  test('it renders className `.list-item` for each `props.items`', () => {
    const component = renderer
      .create(<List items={mockItems} />);

    let instance = component.root;

    const foundClassNames = instance.findAll(
      el => el.props.tag == 'div'
        && el.props.className == 'list-item'
    );

    expect(foundClassNames).toHaveLength(2);
  });

  test('it renders <Loader /> when `props.isFetching` is true', () => {
    const component = renderer
      .create(<List isFetching={true} />);

    let instance = component.root;

    const foundInstance = instance.findAll(
      el => el.type.name == 'Loader'
    );

    expect(foundInstance).toHaveLength(1);
  });

});