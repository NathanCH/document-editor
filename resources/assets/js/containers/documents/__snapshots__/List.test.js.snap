// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> it renders without props 1`] = `
<div
  className="list"
/>
`;
